src/test/scala/battleship/gameActorSpec.scala

```diff
@@ -0,0 +1,136 @@
+package battleship
+
+/**
+  * Created by m06f947 on 3-8-2017.
+  */
+
+import akka.actor.{ActorRef, ActorSystem}
+import akka.testkit.{TestKit, TestProbe}
+import battleship.PlayerActor.{GetNextShot, PlaceBoats}
+import battleship.game.{BoardState, Boat, Game, LinearPlayer}
+import battleship.GameActor.{GameNotStartedYet, _}
+import org.scalatest.{BeforeAndAfterAll, Matchers, WordSpecLike}
+
+import scala.concurrent.duration._
+import scala.language.postfixOps
+
+/**
+  * Created by jordidevos on 28/07/2017.
+  */
+class GameActorSpec(_system: ActorSystem)
+  extends TestKit(_system)
+  with Matchers
+  with WordSpecLike
+  with BeforeAndAfterAll {
+
+    def this() = this(ActorSystem("GameActorSpecActorSystem"))
+
+    override def afterAll: Unit = {
+      shutdown(system)
+    }
+  "gameActor" should {
+    "Play the games given various commands" in {

```

Comment:
This unit test is a too long. A check question you can ask yourself: "_If this test fails, does that well me which part of the tested code is broken?_". In general, a unit test should restrict itself to testing a specific unit of code. Here, you are testing close to the entire GameActor logic. A failing test would tell you that the GameActor is broken, but it won't tell you which part is broken. For you as original author it might be clear, but for someone else modifying your code it will become a time consuming task to figure out what exactly is broken.

Unit testing the internals of an actor is not easy. One way is to refactor much of the logic to a trait external to the actor (see `GameManagerActor` and `GameActorCreator`) but this is not straigtforward when using multiple `context.become` statements. Another way is to use a [TestActorRef](http://doc.akka.io/docs/akka/2.5.3/scala/testing.html#the-way-in-between-expecting-exceptions) to access the internals of the actor. That way you can directly call the functions of type `Receive` and thus directly test the logic in those functions.